agent-control/src/secrets_provider/file.rs

```diff
@@ -0,0 +1,82 @@
+use crate::secrets_provider::SecretsProvider;
+use std::fs;
+use std::path::PathBuf;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+#[error("reading file secret: {0}")]
+pub struct FileSecretProviderError(String);
```

Comment:
I'd use [`std::io::Error`](https://doc.rust-lang.org/std/io/struct.Error.html) directly for this instead of a custom error type. After all we don't do anything outside opening a file.